### ISSUE TITLE
🐛 Bug Report: Accessibility Issue with Keyboard Navigation

### ISSUE THREAD

**Issue by vidhanshah** [backstage/backstage#29452]
### 📜 Description

We are encountering a consistency issue with Keyboard navigation in our documentation site that uses techdocs-core plugin. 
Specifically, when using the keyboard to navigate the left navigation content.

**Plugins**:

- @backstage/plugin-techdocs-backend: 1.11.6
- @backstage/plugin-techdocs-node: 1.13.0
- @backstage/plugin-techdocs: 1.12.3
- @backstage/plugin-techdocs-react: 1.2.14
- @backstage/plugin-techdocs-module-addons-contrib: 1.1.21

### 👍 Expected behavior

It should be able to expand and collapse the folder items and nested folder items from the left navigation. While it is able to expand the folder items (but is not consistent) but not able to collapse.

### 👎 Actual Behavior with Screenshots

When using the keyboard to navigate the left navigation content:

- Pressing Enter or Space does not reliably expand and collapse folder items.
- If the folder contains nested folders, those nested items are not collapsing as expected.
- Also, there seems to be consistency issue as some of the folder items are expanding while few of them are not.

### 👟 Reproduction steps

Sharing the video around what we are observing:

https://github.com/user-attachments/assets/f480c170-65a3-4240-a2fd-8ba4bb0d800e

### 👀 Have you spent some time to check if this bug has been raised before?

- [x] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by awanlin** [comment]
Thanks for logging this @vidhanshah.

Just noting this is easy to reproduce in the Demo site: https://demo.backstage.io/docs/default/component/backstage/overview/what-is-backstage/

**Comment by vidhanshah** [comment]
I have shared the video using the demo site shared above: 

https://github.com/user-attachments/assets/a3eef52d-6430-48ea-915a-20a5a10711f8

**Comment by vidhanshah** [comment]
Hi, I wanted to kindly follow up on this issue. It has been a week since it was reported, and I was wondering if there are any updates or if additional information is needed from my side to help address the problem.


Looking forward to your response. Thank You!!

**Comment by awanlin** [comment]
@vidhanshah, would you be able to submit a fix? Otherwise we'd be looking to the Community to help with this.

**Comment by vidhanshah** [comment]
@awanlin, I am try to look for the nested folder navigation expand/collapse functionality, but I am having trouble locating the file where that specific logic is implemented.
Could you please point me to the file or module where this logic resides? That would help me proceed faster.

Thanks in advance!

**Comment by vidhanshah** [comment]
PR:  https://github.com/backstage/backstage/pull/29960

**Comment by vidhanshah** [comment]
Can I get the PR review for this issue? Reproduce the change as well in the PR as well
